### PR TITLE
Add a sample for DropColumns

### LINKED ISSUE #2353: Add a sample for DropColumns
The `DropColumns` transform doesn't have a sample.

### PR DESCRIPTION
**Pull request by rogancarr**
This PR adds a sample for DropColumns.

Fixes #2353

### CODE REVIEW

File: docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/DropColumns.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            // 42.0   1.0   0-5yrs      1.0         1.0       1.0      2.0  ...
+            // 39.0   1.0   0-5yrs      2.0         6.0       4.0      3.0  ...
+            // 34.0   1.0   0-5yrs      2.0         4.0       2.0      4.0  ...
+            // 35.0   1.0   6-11yrs     1.0         3.0       32.0     5.0  ...
+
+            // DropColumns is commonly used to drop unused columns if the dataset is going to be serialized or
+            // written out to a file. It is not actually necessary to drop unused columns before training or
+            // performing transforms, as IDataView's lazy evaluation won't actually materialize those columns.
+            // In the case of serialization, every column in the schema will be written out. If you have columns
+            // that you don't want to save, you can use DropColumns to remove them from the schema.
```
Comment (sfilipi) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/DropColumns.cs]: Does it make sense to add this info to the remarks section?

Comment (rogancarr) [reply docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/DropColumns.cs]: Yes, moved to remarks.

File: docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/DropColumns.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            // 26.0   1.0   0-5yrs      1.0         6.0       3.0      1.0  ...
+            // 42.0   1.0   0-5yrs      1.0         1.0       1.0      2.0  ...
+            // 39.0   1.0   0-5yrs      2.0         6.0       4.0      3.0  ...
+            // 34.0   1.0   0-5yrs      2.0         4.0       2.0      4.0  ...
+            // 35.0   1.0   6-11yrs     1.0         3.0       32.0     5.0  ...
+
+            // Drop the Age and Education columns from the dataset.
+            var pipeline = mlContext.Transforms.DropColumns("Age", "Education");
+
+            // Now we can transform the data and look at the output
```
Comment (wschin) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/DropColumns.cs]:
```suggestion
            // Now we can transform the data and look at the output.
```

File: src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs [hunk 3]
```diff
@@ -64,8 +64,22 @@ public static ColumnConcatenatingEstimator Concatenate(this TransformsCatalog ca
         /// DropColumns is used to select a list of columns that user wants to drop from a given input. Any column not specified will
         /// be maintained in the output schema.
         /// </summary>
+        /// <remarks>
+        /// DropColumns is commonly used to remove unwanted columns from the schema if the dataset is going to be serialized or
+        /// written out to a file. It is not actually necessary to drop unused columns before training or
+        /// performing transforms, as IDataView's lazy evaluation won't actually materialize those columns.
```
Comment (wschin) [inline src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: Could we have things like `<see cref="IDataView">` for types when possible (an example is [here](https://github.com/dotnet/machinelearning/blob/e5bdd10d78400a51fde9e30846c37837ed40eb01/src/Microsoft.ML.FastTree/RegressionTree.cs#L50))?

File: src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs [hunk 4]
```diff
@@ -64,8 +64,22 @@ public static ColumnConcatenatingEstimator Concatenate(this TransformsCatalog ca
         /// DropColumns is used to select a list of columns that user wants to drop from a given input. Any column not specified will
         /// be maintained in the output schema.
         /// </summary>
+        /// <remarks>
+        /// DropColumns is commonly used to remove unwanted columns from the schema if the dataset is going to be serialized or
```
Comment (wschin) [inline src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: Could we add a reference for DropColumns to `mlContext.Transforms.DropColumns`?

File: src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs (hunk 3 above)
Comment (wschin) [reply src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: Why don't we make a `using`?

**Pull request merged**